### PR TITLE
Jetpack Scan: Don't show '<action> on <date>' if no date is available

### PR DESCRIPTION
**Pull request by mattgawarecki**
#### Changes proposed in this Pull Request

* Add a `Number.isFinite` check to `fixed_on` and `first_detected` API response properties, so that we don't create invalid dates for the UI to display.

#### Testing instructions

* In either Jetpack Cloud or Calypso, select a site that has Jetpack Scan and at least one fixed or ignored threat.
* Navigate to **Jetpack > Scan > History**.
* In React DevTools, check the `status` and `fixedOn` properties of any `ThreatItem` component.
  * Verify that if `fixedOn` is a numeric value, text is displayed to show when action was taken on the threat (see reference screenshots).
  * Verify that if `fixedOn` is undefined or non-numeric, no "<action taken> on <date>" text is displayed.
* Inspect different fixed/ignored threats to ensure behavior is correct in all cases (i.e., fixed and ignored threats, both with and without "fixed on" dates).
* **Alternatively:** In React DevTools, modify the values of a `ThreatItem`'s `status` and `fixedOn` properties to simulate different scenarios.

#### Reference screenshots

Screen Shot 2020-10-07 at 08 40 06
Screen Shot 2020-10-07 at 08 41 58

Screen Shot 2020-10-07 at 08 43 15
Screen Shot 2020-10-07 at 08 43 05

### CODE REVIEW

**Comment by blackjackkent** [comment]
Would it be good to include some sort of text for this scenario but not including a date? (At a quick glance down the example screenshot you provided, my immediate visual interpretation was that the second item was not in the same state as the first since it didn't have that message, even though on a second look I could see they both have the "Fixed" label.)

Maybe just "Threat fixed" or "Resolved" or something like that?

**Review by monsieur-z — APPROVED** [review]
LGTM! I tested the 4 different combinations of statuses and dates.

**Comment by mattgawarecki** [comment]
@blackjackkent Addressing your feedback :-) @monsieur-z Can you verify these messages show up without a date when `threat.fixedOn` isn't defined?

image

image

File: client/components/jetpack/threat-item-subheader/index.tsx [hunk 1]
```diff
@@ -33,9 +33,41 @@ const formatDate = ( date: Date ) => {
...
+			  } );
+	}
+
+	if ( status === 'ignored' ) {
+		return date
+			? translate( 'Threat ignored on %(date)s', {
+					args: { date },
+					comment: 'Past tense action: a threat was ignored on a specific date',
+			  } )
+			: translate( 'Threat ignored', {
```
Comment (a8ci18n) [inline client/components/jetpack/threat-item-subheader/index.tsx:58]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Threat ignored.' )` *ES Score: 13*

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4968162

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4968162).

**Pull request merged**